dev/breeze/doc/adr/0005-preventing-using-contributed-code-when-building-images.md

```diff
@@ -0,0 +1,160 @@
+
+* some code that should be executed "inside" of the images when building should be possible to come from the
+  PR - for example code that is used inside the docker image to install mysql or postgres should be possible
+  to be changed in the PR - as long as we make sure the code is executed inside the Docker image or Docker
+  build process.
+
+Those protections are gradually strengthened (up until recently there was no granular access rights possible)
+however we decided to add certain rules of the "build" code that is executed in our GitHub "Build Images"
+"pull request target" to make sure
+
+## Decision
+
+The decision of our use of GitHub images is to utilise "Pull Request Workflow" to build the shared image,
+but to make sure that the following rules are in-place:
+
+1) We always use `persist-credentials: false` in all GitHub Action checkouts, to prevent unauthorized pushes
+   to our repository
+
+```yaml
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ env.TARGET_COMMIT_SHA }}
+          persist-credentials: false
+          fetch-depth: 2
+```
+
+2) we use submodules (in .github/actions) where we keep actions that we are using (except of the standard
+   GitHub managed actions). Submodules provide few features such as - automated linking to specific commit
+   SHA (not tag) and integration with Pull Request Review process when someone creates a PR to upgrade the
+   action, which makes it ideal to securely and seamlessly keep the action updated if needed.
+
+3) No user code coming from the PR can be executed directly in the "Build image" workflow. For example, the
+   build scripts should not import `setup.py` or execute bash scripts coming from other places than:
+   * scripts/ci
+   * dev/
+   All the other sources are taken from the PR, but those two folders, during the "Build image" workflow
+   are overwritten by the `main` version of the scripts. This means that it is safe to execute those
+   scripts in the "Build Image" workflow.
+
+4) All the other code coming from the PR can only be executed inside the docker container started by the
+   `scripts/ci` or `dev` scripts. The docker containers should not have any volumes mounted from
+   the host that will enable them to read or modify values, environment variables that are present in the
+   Host CI environment.
+
+5) The "docker build" commands automatically execute Dockerfile commands inside such a container, so there
+   is no risk that sensitive information from the host will be passed to them.
+
+6) In case any information is needed from the "sources" of Airflow (such as name of the current branch or
+   version of airflow) it should be extracted by parsing of the incoming scripts but not executing them. Under
+   any circumstances any of the scripts coming from the outside of `scripts/ci` and `dev` should be
+   executed on the host during the image building process.
```

Comment:
Under any circumstances any of the scripts coming from the outside of `scripts/ci` and `dev` should be
   executed on the host during the image building process.

Is the above line correct?

Reply:
Yep. This is advanced security stuff for our 'build' jobs. We really need to make sure that we only execute anything in the build.yml that is coming from one of those two folders only. Because all the rest can be modified by the users making PR and they might get access to stuff we do not want them to.

What we do in our workflows, we make sure that those two folders (for now only scripts/ci) are not coming from the PR of the user but from 'main' branch of our repository

Reply:
@potiuk Under any circumstances any of the scripts coming from the outside of scripts/ci and dev should **not** be executed on the host during the image building process.
Am i right? Not is missing, which will mean you allow any scripts to be executed